src/ytdl_sub/cli/download_args_parser.py

```diff
@@ -61,6 +61,46 @@ def _get_argument_name(cls, arg_name: str) -> str:
         """
         return arg_name[2:]
 
+    @classmethod
+    def _uses_list(cls, argument_name: str) -> bool:
+        """
+        :param argument_name: The argument name which might be using list
+        :return: True if the argument uses list, denoted by ending with '[<positive integer>]'.
+                 False otherwise.
+        """
+        pattern = re.compile(r"\[[1-9]\d*\]$")
+
+        return bool(pattern.search(argument_name))
+
+    @classmethod
+    def _extract_list_index(cls, argument_name: str) -> int:
+        """
+        :param argument_name: The argument name using list
+        :return: Splits the argument and returns the name and the list index.
+        """
+        prefix, _, last = argument_name.rpartition("[")
```

Comment:
nit: you could use a capture group on the regex above, something like:
`r"\[([1-9]\d*)\]$"` - do the match and extraction in one function